compiler/rustc_middle/src/ty/error.rs

```diff
@@ -265,70 +244,18 @@ impl<'tcx> TypeError<'tcx> {
 }
 
 impl<'tcx> Ty<'tcx> {
-    pub fn sort_string(self, tcx: TyCtxt<'_>) -> Cow<'static, str> {
+    pub fn sort_string(self, tcx: TyCtxt<'tcx>) -> String {
```

Comment:
Any reason to drop the `Cow` here?

Reply:
I was fighting with the borrow checker and dropped the Cow in the meantime to avoid misleading diagnostics. After other changes, I think I can get the Cow back.